scripts/sharedFuncs.sh

```diff
@@ -10,16 +10,34 @@ function package_installed() {
         else
             echo "false"
         fi
-    else    
+    else
         if [ "$pkginstalled" -eq 0 ];then
             show_message "package\033[1;36m $1\e[0m is installed..."
         else
-            warning "package\033[1;33m $1\e[0m is not installed.\nplease make sure it's already installed"
-            ask_question "would you continue?" "N"
+            warning "package\033[1;33m $1\e[0m is not installed."
+            ask_question "would you like to install it now?" "Y"
             if [ "$question_result" == "no" ];then
                 echo "exit..."
                 exit 5
-            fi
+            else
+                declare -A osInfo;
+                osInfo[/etc/redhat-release]=yum
+                osInfo[/etc/SuSE-release]=zypper
+                osInfo[/etc/debian_version]=apt-get
+
+                for f in ${!osInfo[@]}
+                do
+                    if [[ -f $f ]];then
+						sudo ${osInfo[$f]} update
+						sudo ${osInfo[$f]} install $1
+						return
+                    fi
+                done
+                # setup for arch-based systems
+            	sudo pacman -Syu
```

Comment:
Maybe this implementation could have some side effects for OSs that don't match with redhat, SuSE, debian and Arch. I think it is better if it prints an error when it is a non tested system. Also, for linux distros there is a file called `/etc/os-release` which has information about the distro and let us do something like this:

```sh
    # Find the current distribution
    if [ -f /etc/os-release ]; then
        if grep -q arch /etc/os-release; then
            os="arch-linux"
        elif grep -q debian /etc/os-release; then
            os="debian"
        elif grep -q void /etc/os-release; then
            os="void-linux"
        elif grep -q alpine /etc/os-release; then
            os="alpine"
        elif grep -q fedora /etc/os-release; then
            os="fedora"
        else
            echo "ERROR: I currently don't have support for your distro"
            exit 1
        fi
    fi
```

Also, to match OS it is possible to do something like this https://github.com/ulises-jeremias/dotfiles/blob/master/util/os.sh

```sh
is_darwin() {
	uname -a | grep Darwin > /dev/null;
}

if [ -z "${os}" ]; then
    # Find the current distribution
    if [ -f /etc/os-release ]; then
        if grep -q arch /etc/os-release; then
            os="arch-linux"
        elif grep -q debian /etc/os-release; then
            os="debian"
        elif grep -q void /etc/os-release; then
            os="void-linux"
        elif grep -q alpine /etc/os-release; then
            os="alpine"
        elif grep -q fedora /etc/os-release; then
            os="fedora"
        else
            echo "ERROR: I currently don't have support for your distro"
            exit 1
        fi
    else
        if is_darwin; then
            os="darwin"
        else
            echo "Error: cannot detect platform"
            exit 1
        fi
    fi
fi

# supported OSs
oss=( arch-linux debian )

if [[ ! " ${oss[*]} " = *"${os}"* ]]; then
    echo "${os} is not supported"
    exit 1
fi
```

Reply:
Anyway, I'll be fine with something like this after the for loop

```sh
if grep -q arch /etc/os-release; then
            # setup for arch-based systems
           sudo pacman -Syu
           sudo pacman -S coreutils
	   sudo pacman -S $
else
           echo "Error: cannot detect platform"
           exit 1
fi
```

Reply:
hi thanks I will test it